### PR TITLE
use grafana authentication

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/roadmap/issues/381

This PR use Grafana authentication mechanism to protect this server endpoints.

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: helm/macropower-analytics-panel-server/templates/ingress-auth.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+apiVersion: networking.k8s.io/v1
```
Comment (QuentinBisson) [inline helm/macropower-analytics-panel-server/templates/ingress-auth.yaml:1]: Ingress v1 is not supported in some installations. I am fine with us using it but we need to feature flag it like it https://github.com/goharbor/harbor-helm/blob/37dc4069bb0b0e03bfa79b4a915d479267a2d247/templates/ingress/ingress.yaml#L31

File: helm/macropower-analytics-panel-server/templates/ingress-auth.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    http:
+      paths:
+      - backend:
+          service:
+            #TODO: template name and port number
+            name: grafana
+            port:
+              number: 3000
+        #TODO: change path
+        path: /grafana-fauth
```
Comment (QuentinBisson) [inline helm/macropower-analytics-panel-server/templates/ingress-auth.yaml:23]: grafana-fauth ?

Comment (QuentinBisson) [reply helm/macropower-analytics-panel-server/templates/ingress-auth.yaml:23]:
```suggestion
        path: /grafana-auth
```

File: helm/macropower-analytics-panel-server/templates/ingress-auth.yaml (hunk 1 above)
Comment (TheoBrigitte) [reply helm/macropower-analytics-panel-server/templates/ingress-auth.yaml:1]: All good I'll revert this.

File: helm/macropower-analytics-panel-server/templates/ingress-auth.yaml (hunk 2 above)
Comment (TheoBrigitte) [reply helm/macropower-analytics-panel-server/templates/ingress-auth.yaml:23]: I do not know what to put in there yet, `/grafana-auth`, `/analytics-plugin/auth` ? unsure yet

**Review by QuentinBisson — APPROVED** [review]
Not a big fan of todos but I'd rather not block you there

File: helm/macropower-analytics-panel-server/templates/ingress-auth.yaml (hunk 2 above)
Comment (QuentinBisson) [reply helm/macropower-analytics-panel-server/templates/ingress-auth.yaml]: If it's only for the plugin, I would go with /analytics-plugin/auth or login yes :)

**Pull request merged**